superduperdb/components/component.py

```diff
@@ -114,11 +117,19 @@ def on_load(self, db: Datalayer) -> None:
 
     def dict(self) -> 'Document':
         from superduperdb import Document
+        from superduperdb.components.datatype import Artifact, File
 
         r = Document(super().dict())
         s = self.artifact_schema
         for k in s.fields:
-            r[f'dict.{k}'] = s.fields[k](getattr(self, k))  # artifact
+            attr = getattr(self, k)
+            if isinstance(attr, (Artifact, File)):
```

Comment:
Just a nitpick: can we create a LazyComponent class and subclass Artifact and File from it 
and do isintance (attr, LazyComponent) project wide ?
so that later if another lazy component like artifact or file comes into picture we just subclass it and be done with it.

Reply:
I created a base class for `Artifact` now.